components/automate-load-balancer/habitat/config/nginx.conf

```diff
@@ -145,6 +145,12 @@ http {
     #
     add_header Cache-Control "no-cache, no-store, must-revalidate";
 
+    # Implement secure headers in accordance w/ owasp best practices
+    # https://www.owasp.org/index.php/OWASP_Secure_Headers_Project#tab=Headers
+    add_header x-xss-protection "1; mode=block" always;
+    add_header Strict-Transport-Security "max-age=63072000; includeSubDomains" always;
+    add_header X-Content-Type-Options "nosniff" always;
```

Comment:
❓ Do we need to copy these around like the `Cache-Control` headers, following this comment? https://github.com/chef/automate/blob/23fdf23efeaa27355e3072f827b0024fde90c0d7/components/automate-load-balancer/habitat/config/nginx.conf#L142-L145

Reply:
As these lines were added to a `server` block, I don't believe this would apply.

Reply:
I'm not claiming that this wasn't true but I suppose by that logic, we should be able to drop all those `add_header Cache-Control "...";` statements below then. 🤔

Reply:
How would you prefer to proceed? I'm open to moving these lines above the Cache-Control line on line 146 if that would help, or I could move limit the scope to the location for automate-ui, which also has its own Cache-Control.

Reply:
The comment here is still valid. Consider:

```
[13][default:/src:0]# curl -kv https://localhost/api/ 2>&1 >/dev/null | grep '^<'
< HTTP/2 404 
< server: nginx
< date: Tue, 09 Jul 2019 10:33:23 GMT
< content-type: text/plain; charset=utf-8
< content-length: 10
< x-content-type-options: nosniff
< x-xss-protection: 1; mode=block
< strict-transport-security: max-age=63072000; includeSubDomains
< x-content-type-options: nosniff
< 
[14][default:/src:0]# curl -kv https://localhost/dex/ 2>&1 >/dev/null | grep '^<'
< HTTP/2 404 
< server: nginx
< date: Tue, 09 Jul 2019 10:33:25 GMT
< content-type: text/plain; charset=utf-8
< content-length: 19
< x-content-type-options: nosniff
< 
```

Thus, if we want these headers universally applied (I think we do?) then we need to copy them into any location block that uses `add_header`.